### PR TITLE
Bump: Minimum Python version 3.10

### PR DESCRIPTION
**Pull request by ClausHolbechArista**
Bump: Minimum Python version 3.10 incl update of docs and removal of special 3.9 workarounds.

### CODE REVIEW

File: ansible_collections/arista/avd/docs/release-notes/5.x.x.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+- Documentation for AVD version `5.x.x` [available here](https://avd.arista.com/devel/)
+
+## Release 5.0.0
+
+### Changes to requirements
+
+- AVD v.5.0.0 requires Python version 3.10 or newer.
+
+TODO
```
Comment (gmuloc) [inline ansible_collections/arista/avd/docs/release-notes/5.x.x.md]:
```suggestion
```

File: ansible_collections/arista/avd/plugins/action/verify_requirements.py [hunk 2]
```diff
@@ -26,7 +26,8 @@
     HAS_PACKAGING = False
 
 # Python >= 3.9
```
Comment (gmuloc) [inline ansible_collections/arista/avd/plugins/action/verify_requirements.py]:
```suggestion
```

**Review by gmuloc — CHANGES_REQUESTED** [review]

File: ansible_collections/arista/avd/docs/release-notes/5.x.x.md (hunk 1 above)
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/docs/release-notes/5.x.x.md]:
```suggestion
```

**Review by gmuloc — APPROVED** [review]
:shipit:

**Review by carlbuchmann — APPROVED** [review]
LGTM tested with Python 3.9 and correct errors are raised

**Pull request merged**